src/main/scala/mesosphere/marathon/api/akkahttp/v2/TasksController.scala

```diff
@@ -108,8 +167,51 @@ class TasksController(
     case _ => None
   }
 
+  /**
+    * Performs the task kill on the provided taskIds (without scale). Delegates the job to TaskKiller.
+    * @return list of killed tasks
+    */
+  @SuppressWarnings(Array("all")) // async/await
+  private def kill(toKillFuture: Future[Map[PathId, Seq[Instance]]], tasksIdToAppId: Map[Id, PathId], wipe: Boolean)(implicit identity: Identity): Future[Seq[EnrichedTask]] = async {
+    val toKill = await(toKillFuture)
+    val affectedApps = tasksIdToAppId.values.flatMap(appId => groupManager.app(appId))(collection.breakOut)
+
+    val killedTasks = await(Future.sequence(toKill
+      .filter { case (appId, _) => affectedApps.exists(app => app.id == appId) }
+      .map { case (appId, instances) => taskKiller.kill(appId, _ => instances, wipe) }))
+      .flatten
+
+    killedTasks.flatMap { instance =>
+      instance.tasksMap.valuesIterator.map { task =>
+        EnrichedTask(task.runSpecId, task, instance.agentInfo, Seq.empty)
```

Comment:
Haven't we introduce a breaking change in the API here?

Reply:
Just if someone finds this comment - it looks like apps/delete returns a different json than /delete ... so this change is fine, but the API sucks :)